ports/nrf/common-hal/_bleio/Adapter.c

```diff
@@ -386,6 +386,15 @@ bleio_address_obj_t *common_hal_bleio_adapter_get_address(bleio_adapter_obj_t *s
     return address;
 }
 
+uint32_t common_hal_bleio_adapter_set_address(bleio_adapter_obj_t *self, bleio_address_obj_t *address)
+{
+    ble_gap_addr_t local_address;
+    local_address.addr_type = address->type;
+    const char *data = mp_obj_str_get_str(address->bytes);
+    memcpy(local_address.addr, data, NUM_BLEIO_ADDRESS_BYTES);
```

Comment:
Use this code instead to get a pointer to the bytes. It's the standard way of getting the length and a pointer to the bytes in some kind of bytes-like object.
```c
    mp_buffer_info_t bufinfo;
    if (!mp_get_buffer(address->bytes, &bufinfo), MP_BUFFER_READ) {
        return false;
    }
    memcpy(local_address.addr, bufinfo.buf, NUM_BLEIO_ADDRESS_BYTES);
```